llms/speculative_decoding/prompt_lookup.py

```diff
@@ -0,0 +1,100 @@
+import argparse
+import time
+
+import mlx.core as mx
+from decoder import PromptLookupDecoder
+from mlx.utils import tree_unflatten
+from model import Model
+from transformers import T5Config
+
+
+def load_model(model_name: str):
+    config = T5Config.from_pretrained(model_name)
+    model = Model(config)
+    weights = mx.load(f"{model_name}.npz")
+    weights = tree_unflatten(list(weights.items()))
+    model.update(weights)
+    mx.eval(model.parameters())
+    return model
```

Comment:
Let's refactor this into `Model` since they both use the exact same functionality?